src/macaron/slsa_analyzer/git_url.py

```diff
@@ -138,16 +141,10 @@ def check_out_repo_target(git_obj: Git, branch_name: str = "", digest: str = "")
 
     logger.info("Successfully checked out branch %s.", res_branch)
 
-    try:
-        head_commit: Commit = git_obj.repo.rev_parse("HEAD")
-    # https://gitpython.readthedocs.io/en/stable/reference.html?highlight=rev_parse#git.repo.base.Repo.rev_parse
-    except (BadObject, IndexError, ValueError) as error:
-        logger.critical("Cannot get the head commit after checking out. Error %s.", error)
-        return False
-
-    # We only pull the latest changes if no digest is provided
-    # or a commit digest is provided but it does not exist in the current local branch.
-    if not digest or (digest and not commit_exists(git_obj, digest)):
+    # We only pull the latest changes if we are not running in offline mode and:
+    #   - no digest is provided.
+    #   - or a commit digest is provided but it does not exist in the current local branch.
+    if not offline_mode and (not digest or (digest or not commit_exists(git_obj, digest))):
```

Comment:
Why do you check `digest` again? We get to the second expression in the `or` statement only when `not digest` is `False`.

Reply:
I agree. It could be removed as you mentioned. 
`if not offline_mode and (not digest or not commit_exists(git_obj, digest)):
`

Reply:
Changed in https://github.com/oracle-samples/macaron/pull/125/commits/e1e76d703e2bc1fda48d40537f76ede910421796.